moto/iam/config.py

```diff
@@ -0,0 +1,321 @@
+import json
+import boto3
+from moto.core.exceptions import InvalidNextTokenException
+from moto.core.models import ConfigQueryModel
+from moto.iam import iam_backends
+
+
+class RoleConfigQuery(ConfigQueryModel):
+    def list_config_service_resources(
+        self,
+        resource_ids,
+        resource_name,
+        limit,
+        next_token,
+        backend_region=None,
+        resource_region=None,
+        aggregator=None,
+    ):
+        # IAM roles are "global" and aren't assigned into any availability zone
+        # The resource ID is a AWS-assigned random string like "AROA0BSVNSZKXVHS00SBJ"
+        # The resource name is a user-assigned string like "MyDevelopmentAdminRole"
+        # Stored in moto backend with the AWS-assigned random string like "AROA0BSVNSZKXVHS00SBJ"
+
+        # Grab roles from backend; need the full values since names and id's are different
+        role_list = list(self.backends["global"].roles.values())
+
+        if not role_list:
+            return [], None
+
+        # Filter by resource name or ids
+        if resource_name or resource_ids:
+            filtered_roles = []
+            # resource_name takes precedence over resource_ids
+            if resource_name:
+                for role in role_list:
+                    if role.name == resource_name:
+                        filtered_roles = [role]
+                        break
+                # but if both are passed, it must be a subset
+                if filtered_roles and resource_ids:
+                    if filtered_roles[0].id not in resource_ids:
+                        return [], None
+            else:
+                for role in role_list:
+                    if role.id in resource_ids:
+                        filtered_roles.append(role)
+
+            # Filtered roles are now the subject for the listing
+            role_list = filtered_roles
+
+        if aggregator:
+            # IAM is a little special; Roles are created in us-east-1 (which AWS calls the "global" region)
+            # However, the resource will return in the aggregator (in duplicate) for each region in the aggregator
+            # Therefore, we'll need to find out the regions where the aggregators are running, and then duplicate the resource there
+
+            # In practice, it looks like AWS will only duplicate these resources if you've "used" any roles in the region, but since
+            # we can't really tell if this has happened in moto, we'll just bind this to the regions in your aggregator
+            aggregated_regions = []
+            aggregator_sources = aggregator.get(
+                "account_aggregation_sources"
+            ) or aggregator.get("organization_aggregation_source")
+            for source in aggregator_sources:
+                source_dict = source.__dict__
+                if source_dict["all_aws_regions"]:
```

Comment:
Will this always be present in the dictionary? I.e. should this use `source_dict.get("all_aws_regions")` instead?